sail/src/main/java/org/apache/rya/rdftriplestore/inference/InferenceEngine.java

```diff
@@ -762,6 +800,57 @@ public void handleStatement(final Statement statement) throws RDFHandlerExceptio
     }
 
     /**
+     * For a given type, return any properties and values such that owl:hasSelf restrictions on
+     * those properties could imply this type. No matter how many restrictions are returned, each
+     * one is considered individually sufficient: if a resource has the property and the value, then
+     * it belongs to the provided type. Takes type hierarchy into account, so the value may imply a
+     * subtype which in turn implies the provided type.
+     * @param type The type (URI or bnode) to check against the known restrictions
+     * @return For each relevant property, a set of values such that whenever a resource has that
+     *      value for that property, it is implied to belong to the type.
+     */
+    public Set<URI> getHasSelfImplyingType(final Resource type){
+        // return properties that imply this type if reflexive
+        final Set<URI> properties = new HashSet<>();
+        properties.addAll(hasSelfByType.get(type));
```

Comment:
This line can cause an NPE if there is no hasSelf property associated with the type. Looks like that's why unrelated tests can fail if the HasSelfVisitor gets a chance to visit them -- any statement pattern like "?x rdf:type :T" will end up here, so it needs to check whether hasSelf logic applies before proceeding.

Reply:
merp.  ok I'll add null checks to both functions